drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -194,45 +459,39 @@ static int eth_esp32_dev_init(const struct device *dev)
 
 	k_thread_name_set(tid, "esp_event");
 
+	if (IS_ENABLED(CONFIG_ESP32_WIFI_STA_AUTO_DHCPV4)) {
+		net_mgmt_init_event_callback(&esp32_dhcp_cb, wifi_event_handler, DHCPV4_MASK);
+		net_mgmt_add_event_callback(&esp32_dhcp_cb);
+	}
+
 	wifi_init_config_t config = WIFI_INIT_CONFIG_DEFAULT();
 	esp_err_t ret = esp_wifi_init(&config);
 
-	ret |= esp_supplicant_init();
+	ret |= esp_wifi_set_mode(WIFI_MODE_STA);
 	ret |= esp_wifi_start();
 
-	if (IS_ENABLED(CONFIG_ESP32_WIFI_STA_AUTO)) {
-		wifi_config_t wifi_config = {
-			.sta = {
-				.ssid = CONFIG_ESP32_WIFI_SSID,
-				.password = CONFIG_ESP32_WIFI_PASSWORD,
-			},
-		};
-
-		ret = esp_wifi_set_mode(WIFI_MODE_STA);
-		ret |= esp_wifi_set_config(ESP_IF_WIFI_STA, &wifi_config);
-		ret |= esp_wifi_connect();
-	}
-
 	if (ret != ESP_OK) {
-		LOG_ERR("Connect failed");
+		LOG_ERR("Wifi driver failed");
 	}
 
 	return ret;
 }
 
-
-static struct esp32_wifi_runtime eth_data;
-
-static const struct ethernet_api eth_esp32_apis = {
-	.iface_api.init	= eth_esp32_init,
-	.send =  eth_esp32_send,
+static const struct net_wifi_mgmt_offload esp32_api = {
+	.wifi_iface.iface_api.init = esp32_wifi_init,
+	.wifi_iface.send = esp32_wifi_send,
```

Comment:
shouldn't this also be protected with CONFIG_WIFI_OVER_ETHERNET_IFACE ?

Reply:
Well, `esp_wifi_drv.c` is only added into build when ESP32+WIFI is select. Otherwise, I see no need to protect it. Do you?

Reply:
Ah that is auto-selected in Kconfig, should be fine, if someone removes that it will throw build error anyway.